gridt/src/app/core/settings.service.ts

```diff
@@ -22,27 +24,98 @@ export class SettingsService {
     SETIDFAIL: "Could not SET identity",
     NOTLOGGEDIN: "Not logged in",
     SECSTOREUNAVAILABLE: "Secure storage not available.",
-  }
+  };
+
+  public empty_identity: Identity = {
+    id: null,
+    username: "",
+    bio: "",
+    email: "",
+    avatar: ""
+  };
+
+  /**
+  * Emits the last known identity
+  */
+  private _user_identity$ = new ReplaySubject<Identity>(1);
+
+  /**
+  * This Observable can be used in applications to display the latest known
+  * identity of the user.
+  */
+  public userIdentity$: Observable<Identity> = this._user_identity$.asObservable();
+
+  /**
+  * This function updates the user identity.
+  * - If both server and local identity are unavailable, emit empty ID.
+  * - When server ID is retreived, but local ID has not been set yet, store
+  *   server ID in local and emit local ID.
+  * - When the server ID is different from local ID, overwrite local ID with
+  *   the server ID. Then emit local ID.
+  * - If both the server and local identity agree, emit local ID.
+  * - If the server is unavailable but local ID is available, emit local ID.
+  * - Only when updateIdentity() function is called should the userIdentity$
+  *   change.
+  *
+  * - Makes sure all errors are caught such that this function always gives an
+  *   output.
+  */
+  private mapErrorToEmptyID = pipe(
+    catchError( err => {console.warn(err); return of<Identity>(this.empty_identity);}),
+    flatMap( (id) => of(<Identity>(id)))
+  );
+
+  private mapErrorToFalse = pipe(
+    catchError( err => {console.warn(err); return of<boolean>(false);})
+  );
+
+  public updateIdentity(): void {
+    // Combine server and local id. Catch errors to make sure pipe runs.
+    forkJoin({
+      server: this.api.userIdentity$().pipe( this.mapErrorToEmptyID ),
+      local: this.localIdentity$.pipe( this.mapErrorToEmptyID )
+    }).pipe(
+      // Update local id when server id is different.
+      tap( (ids) => console.debug(`recieved server id = ${JSON.stringify(ids.server)} \n received local id = ${JSON.stringify(ids.local)}`)),
+      flatMap( (ids) => iif(
+        () => ((JSON.stringify(ids.server)==JSON.stringify(ids.local)) || (JSON.stringify(ids.server) == JSON.stringify(this.empty_identity))), // && ids[0].id == ids[1].id
```

Comment:
should these not be triple =? and check for spacing around the ==

Reply:
line 81 is also quite long, i would suggest splitting it into a new line after the ||

Reply:
Also: you leave commented code at the end of this line. Please remove it before merger.